staging/src/k8s.io/apimachinery/pkg/runtime/converter.go

```diff
@@ -358,38 +497,57 @@ func pointerFromUnstructured(sv, dv reflect.Value) error {
 	dv.Set(reflect.New(dt.Elem()))
 	switch st.Kind() {
 	case reflect.Ptr, reflect.Interface:
-		return fromUnstructured(sv.Elem(), dv.Elem())
+		return fromUnstructured(sv.Elem(), dv.Elem(), ctx)
 	default:
-		return fromUnstructured(sv, dv.Elem())
+		return fromUnstructured(sv, dv.Elem(), ctx)
 	}
 }
 
-func structFromUnstructured(sv, dv reflect.Value) error {
+func structFromUnstructured(sv, dv reflect.Value, ctx *fromUnstructuredContext) error {
 	st, dt := sv.Type(), dv.Type()
 	if st.Kind() != reflect.Map {
 		return fmt.Errorf("cannot restore struct from: %v", st.Kind())
 	}
 
+	pathLen := len(ctx.parentPath)
+	defer func() {
+		ctx.parentPath = ctx.parentPath[:pathLen]
+	}()
+	ctx.pushMatchedKeyTracker()
```

Comment:
do the inline check here (rather than in pushMatchedKeyTracker) so the inline logic is clear

```suggestion
if !origInlined {
	ctx.pushMatchedKeyTracker()
}
```

up to you, but a name like `svInlined := ctx.isInlined` might be clearer than `origInlined`